File: go-gin/internal/modules/contact/contract.go
package contact

type CreateContactDto struct {
	Name  string `json:"name"`
	Email string `json:"email"`
	Phone string `json:"phone"`
}

type UpdateContactDto struct {
	ID     string `json:"id"`
	UserID string `json:"user_id"`
	Name   string `json:"name"`
	Email  string `json:"email"`
	Phone  string `json:"phone"`
}

type Contact struct {
	ID     string `json:"id"`
	UserID string `json:"user_id"`
	Name   string `json:"name"`
	Email  string `json:"email"`
	Phone  string `json:"phone"`
}

File: nestjs/src/modules/contact/contract.ts
export interface Contact {
  id: string;
  name: string;
  email: string;
  phone: string;
}

export interface CreateContactDto {
  name: string;
  email: string;
  phone: string;
}

export interface UpdateContactDto {
  name?: string;
  email?: string;
  phone?: string;
}

export interface CreateContactResponse {
  
}